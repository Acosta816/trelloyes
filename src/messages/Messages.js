import React, {Component} from 'react';
import './Messages.css';

/*let's explain const bubble. So here it's saying, if the expression "props.unread" is equal to false, 
then const bubble is <div id="wow"...</div>, ELSE const bubble = value of null (which is no value at all, 
therefore, nothing will display. The div will never be created either.)
WARNING: If you pass 0 into unread from the app, the expression will be false because in JS, 0  has a falsey value and is
parsed into a boolean false. Therefore the null section of the ternary operator will be called and bubble will equal null.
It still shows up on the React chrome-tools as a child, just has a value of null and so is not displayed obviously because it's never assigned a <tag />  */  
function Messages(props) {
    const bubble = props.unread ?
    <div  id="wow" className = "unread_count">{props.unread}</div> : null;

    return (
        <div className="messages">
            {props.name}
            {bubble}
        </div>
    );
}

export default Messages;