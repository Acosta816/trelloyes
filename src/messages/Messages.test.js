import  React, {Component} from 'react';
import ReactDOM from 'react-dom';
import Renderer from 'react-test-renderer';
import Messages from './Messages';




describe('Test Suite for Messges.js', () => {

    it('Smoke: renders without crashing', () => {
        const div = document.createElement('div');

        ReactDOM.render(<Messages unread={100} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Snapshot#1: renders UI as expected ', () => {
        const messagesJson = Renderer.create(<Messages name="Inbox" unread={100} />).toJSON();
        expect(messagesJson).toMatchSnapshot();
    });

    it('Snapshot#2: Renders UI as expected with unread={0}', () => {
        const messagesJson2 = Renderer.create(<Messages name="Mail" unread={0} />).toJSON();
        expect(messagesJson2).toMatchSnapshot();
    }); 


 /*Snapshot renders the component UI and turns it into JSON, 
 then compares that JSON-->( ..renderer.create(<.../>).toJSON() )  
 to a photo take prior that we have on file--> (.. .toMatchSnapshot().. )   
 and that's all it does! */
});