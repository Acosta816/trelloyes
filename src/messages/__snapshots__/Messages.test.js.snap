// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Suite for Messges.js Snapshot#1: renders UI as expected  1`] = `
<div
  className="messages"
>
  Inbox
  <div
    className="unread_count"
    id="wow"
  >
    100
  </div>
</div>
`;

exports[`Test Suite for Messges.js Snapshot#2: Renders UI as expected with unread={0} 1`] = `
<div
  className="messages"
>
  Mail
</div>
`;

exports[`Test Suite for Messges.js Snapshot: renders UI as expected  1`] = `
<div
  className="messages"
>
  Inbox
  <div
    className="unread_count"
    id="wow"
  >
    100
  </div>
</div>
`;
